specs/BrowserProcessExited.md

```diff
@@ -66,19 +78,33 @@ void AppWindow::CloseWebView(/* ... */) {
               ICoreWebView2Environment* sender,
               ICoreWebView2BrowserProcessExitedEventArgs* args) {
           COREWEBVIEW2_BROWSER_PROCESS_EXIT_KIND kind;
+          UINT32 pid;
           CHECK_FAILURE(args->get_BrowserProcessExitKind(&kind));
-
-          // Watch for graceful browser process exit. Let ProcessFailed event
-          // handler take care of failed browser process termination.
-          if (kind == COREWEBVIEW2_BROWSER_PROCESS_EXIT_KIND_NORMAL)
+          CHECK_FAILURE(args->get_BrowserProcessId(&pid));
+
+          // If a new WebView is created from this CoreWebView2Environment after
+          // the browser has exited but before our handler gets to run, a new
+          // browser process will be created and lock the user data folder
+          // again. Do not attempt to cleanup the user data folder in these
+          // cases. We check the PID of the exited browser process against the
+          // PID of the browser process to which our last CoreWebView2 attached.
+          if (pid == m_newestBrowserPid)
```

Comment:
We don't guarantee that the pid isn't reused?

https://devblogs.microsoft.com/oldnewthing/20110107-00/?p=11803

Reply:
We don't, but it can't be the same for the case for which PID might be relevant: if the environment instance creates a webview and connects to a new browser process after an older one exited but before the handler got to run.

Reply:
I mean, can the old browser process exit, our event is queued, a new browser process starts with the same PID as the old one, and then our event handler gets to run? In that case checking if pid == m_newestBrowserPid doesn't help? PIDs are only uniquely identifying when the process is running

Reply:
The new browser process can't start with the same PID because the environment instance tracking the old process keeps a handle open and so the process object stays valid; it's PID can't be taken until the environment instance observing it closes the handle, which happens only after the queued event gets processed. From the linked post above:

> The process object remains as long as the process is still running (the process implicitly retains a reference to itself) or as long as somebody still has a handle to the process object.

Reply:
You mean the process is gone but we keep a handle to it open so that the PID remains taken? If so, great!

Reply:
Yes, exactly like that.